apps/prairielearn/src/sync/fromDisk/assessments.sql

```diff
@@ -26,17 +26,20 @@ FROM
 SELECT
   q.qid,
   q.id,
-  c.sharing_name
+  qc.sharing_name
 FROM
   questions AS q
-  JOIN sharing_set_questions AS ssq ON q.id = ssq.question_id
-  JOIN sharing_sets AS ss ON ssq.sharing_set_id = ss.id
-  JOIN sharing_set_courses AS ssc ON ss.id = ssc.sharing_set_id
-  JOIN pl_courses AS c ON c.id = ss.course_id
-  JOIN jsonb_to_recordset($imported_question_info::JSONB) AS iqi (sharing_name text, qid text) ON iqi.sharing_name = c.sharing_name
-  AND iqi.qid = q.qid
+  LEFT JOIN pl_courses AS qc on qc.id = q.course_id
```

Comment:
We actually need to join the course onto all the rows so that we can check the course sharing name against the `@sharing-name` in the JSON.  I'm fairly confident that the query planner will do something smart given how simple and clear the where condition is `ssc.course_id = $course_id OR q.shared_publicly;` What do you think @mwest1066 ?

If we wanted to try to do something smart here we, would need to join the course onto the publicly shared questions, then separately also join the course onto the questions shared by sharing set. I'm afraid this will make the query confusing to read.

Reply:
Queries with OR are not handled well in Postgres at all. Best option is to make a union of two queries. See https://github.com/PrairieLearn/PrairieLearn/pull/7381#discussion_r1153515424 (including discussion).

Reply:
Thanks for the wisdom jonathon. I can rewrite this with a Union in a pretty nice way, that should be good.

Reply:
All done. 

The query is rather long now, but the UNION does make it very clear what is going on. 

One part for sharing sets, one part for public sharing, unioned together.